plugin/src/main/java/io/fabric8/maven/plugin/mojo/build/ResourceMojo.java

```diff
@@ -174,10 +174,16 @@
      * The OpenShift deploy timeout in seconds:
      * See this issue for background of why for end users on slow wifi on their laptops
      * DeploymentConfigs usually barf: https://github.com/openshift/origin/issues/10531
+     *
+     * Lets default to 2 hours which should be plenty of time on a slow connection to download
+     * all the images for a YAML/JSON manifest (which can contain many separate images)
      */
-    @Parameter(property = "fabric8.openshift.deployTimeoutSeconds")
+    @Parameter(property = "fabric8.openshift.deployTimeoutSeconds", defaultValue = "7200")
```

Comment:
If I remember our discussions at that time (along with @jimmidyson and other) we had severe issues with ever hanging deployments which are not due to a slow wifi but because of an real error. 

So this is really a trade off. 

As it happens on deployment time, there is no need to add everything on the pom.xm but if you are on a slow wifi connection you always can add the timeout property on the command line. Tbh, I wouldn't try to be cleverer than OpenShift itself here and rely on the standard OpenShift timeouts. Or asked otherwise: How does `oc apply` reacts in this situations ?

And then, how often happens that you are on a slow wifi (probably mostly only for evangelists travelling around) vs. those who have decent connections (which the would be annoyed because of endlessly hanging deployments).

As I said, that's a trade off and we already had this discussion (and a decision). But we can do it again (wonder though what has changed since that time).